src/messaging/ExchangeContext.cpp

```diff
@@ -499,5 +510,22 @@ void ExchangeContext::MessageHandled()
     Close();
 }
 
+bool ExchangeContext::IsValidGroupMsgType(Protocols::Id protocolId, uint8_t msgType)
```

Comment:
This is really odd to have the ExchangeContext (which is not supposed to understand the specifics of protocols) validate this.

Validation of this should happen in the actual protocol logic, and not here.

Reply:
The sooner we check and discard the better it is. However if you've a specific location where this function should be I'll gladly move it there. However since we're on a tight schedule for the group feature I would leave it there for the time being.

Reply:
The logic should be in the various IM clients (i.e `ReadClient`, `CommandSender`, `WriteClient`) to validate whether groupcast messaging is permitted for those actions. Those will have to do more than just allow/disallow group comms - they have to actually validate the paths embedded in the request are compliant to what is expected of a groupcast exchange (i.e all paths have to target a GroupId and should not contain an Endpoint ID in them).

Given that, I'd rather we just remove this bit of logic from this file for now, and add the actual checks needed in the requisite IM objects as follow-ups.